### PR TITLE
Refactor UnderscoreToStudlyCase Filter

### PR DESCRIPTION
**Pull request by ramchale**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Refactoring UnderscoreToStudlyCase based on https://github.com/laminas/laminas-filter/issues/177

### CODE REVIEW

File: src/Word/UnderscoreToStudlyCase.php [hunk 1]
```diff
@@ -4,50 +4,41 @@
...
     public function filter(mixed $value): mixed
     {
         if (! is_scalar($value) && ! is_array($value)) {
             return $value;
         }
 
         /** @var string|array $value */
-        $value          = parent::filter($value);
-        $lowerCaseFirst = 'lcfirst';
+        $value = (new SeparatorToCamelCase('_'))($value);
```
Comment (gsteel) [inline src/Word/UnderscoreToStudlyCase.php]: Nit picky, but could you call `->filter($value)` here please?

File: docs/book/v3/word.md [hunk 2]
```diff
@@ -323,3 +323,22 @@ print $filter->filter('this_is_my_content');
...
+
+### Basic Usage
+
+```php
+$filter = new Laminas\Filter\Word\UnderscoreToStudlyCase();
+
+print $filter->filter('this_is_my_content');
+```
+
+The above example returns `thisIsMyContent`.
```
Comment (gsteel) [inline docs/book/v3/word.md:344]: 👌

**Review by gsteel — CHANGES_REQUESTED** [review]
Thanks @ramchale - Just one small nit pick :)

File: src/Word/UnderscoreToStudlyCase.php (hunk 1 above)
Comment (ramchale) [reply src/Word/UnderscoreToStudlyCase.php]: No problem. I've amended that

**Review by gsteel — APPROVED** [review]
Thank you @ramchale

**Comment by ramchale** [comment]
No worries 😄

**Pull request merged**